taxonomy/providers/course_metadata.py

```diff
@@ -26,8 +26,23 @@ def get_courses(self, course_ids):
           list<dict>: A list of courses in the form of dictionary.
             Dictionary object must have the following keys
             1. uuid: Course UUID
-            1. key: Course key
-            1. title: Course Title
-            1. short_description: Course's short description
-            2. full_description: Course's full description
+            2. key: Course key
+            3. title: Course Title
+            4. short_description: Course's short description
+            5. full_description: Course's full description
+        """
+
+    @abstractmethod
+    def get_all_courses(self):
```

Comment:
Please add validation logic for this new method in `taxonomy/validators/course_metadata.py`

Reply:
Also, add tests for the new method in `tests/test_validators.py`

Reply:
on it.